src/panels/UserPanelsPanel/UserPanelsPanel.tsx

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from 'react';
+import {
+  Button,
+  Container,
+  Divider,
+  Group,
+  Select,
+  Text,
+  TextInput,
+  Title
+} from '@mantine/core';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { UserPageIcon } from '@/icons/icons';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import { intializeUserPages } from '@/redux/userpages/userpagesSlice';
+import { useWindowManagerProvider } from '@/windowmanagement/WindowLayout/hooks';
+
+import { UserPage } from './UserPanel';
+
+export function UserPanelsPanel() {
+  const { addWindow } = useWindowManagerProvider();
+
+  const [selectedPanel, setSelectedPanel] = useState<string | null>(null);
+  const [panelURL, setPanelURL] = useState<string>('');
+  const [urlPanelTitle, setUrlPanelTitle] = useState<string>('');
+
+  const luaApi = useOpenSpaceApi();
+  const isDataInitialized = useAppSelector((state) => state.userPages.isInitialized);
+
+  const localPanels = useAppSelector((state) => state.userPages.panels || []);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    // Collect all folder paths in the USER folder
+    const getUserPanels = async () => {
+      if (!luaApi) return;
+
+      // eslint-disable-next-line no-template-curly-in-string
+      const root = await luaApi.absPath('${USER}/webpanels');
+      const folders = await luaApi.walkDirectoryFolders(root);
+
+      if (!folders) return;
+
+      const slash = navigator.platform.indexOf('Win') > -1 ? '\\' : '/';
+      const folderNames = Object.values(folders).map((panel) =>
+        // Get the folder name from the path
+        panel.substr(panel.lastIndexOf(slash) + 1)
```

Comment:
Will this really work when the slash is the `'\\'` version?  (thinking of the +1)

Maybe:
```suggestion
        panel.split(slash).pop()
```
?

Reply:
Here I also get a deprecation warning for `substr`, so using that version of the function should probably also be avoided. If you want to keep using that version of the code, replace it with `substring`

![image](https://github.com/user-attachments/assets/cb3a42a0-37a4-4663-8e2c-2ace479d7129)

Reply:
It will work - clarified the code a bit